src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -4558,10 +4558,12 @@ def _module_handler(self, node, parent):
         try:
             spec_part = _first_type_match(
                 node.children, Fortran2003.Specification_Part)
+        except ValueError:
+            spec_part = None
+
+        if spec_part is not None:
```

Comment:
Is this fix needed because we could end up with `spec_part == None`, even if no `ValueError` is raised? Now that I look at the modified tests, I see that it's so we don't silently ignore a ValueError raised by `process_declarations`?